WalletWasabi/Wallets/WalletFilterProcessor.cs

```diff
@@ -344,8 +344,14 @@ private async Task<bool> ProcessFilterModelAsync(FilterModel filter, SyncType sy
 					return true;
 				}
 
-				if (toTestKeys.Count == prefetchResult.KeysTested.Count)
+				if (toTestKeys.Count <= prefetchResult.KeysTested.Count)
 				{
+					if (toTestKeys.Count < prefetchResult.KeysTested.Count)
+					{
+						// TODO: I've seen this case happening and this seems to be a bug,
+						// I don't understand any logic where it makes sense to be in that case.
```

Comment:
So this is what happens: https://github.com/zkSNACKs/WalletWasabi/pull/12148/files#r1437044450